lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -177,7 +214,14 @@ def main():
         else:
             module.fail_json(msg="unable to find %s. Exception message: %s" % (config_file, to_native(e)))
 
-    mysqlvar_val = getvariable(cursor, mysqlvar)
+    mysqlvar_val = None
+    if mode == 'global' or mode == 'persist':
+        mysqlvar_val = getvariable(cursor, mysqlvar)
+    elif mode == 'persist_only':
+        mysqlvar_val = check_mysqld_auto(cursor, mysqlvar)
+        if mysqlvar_val is None:
+            mysqlvar_val = getvariable(cursor, mysqlvar)
```

Comment:
I would remove the previous and the current line. If you write `persist_only`, you want to make sure that the variable is set in he `mysqld_auto` config. If it is not set there (but it is set in the running instance), you still want to execute the query.

Reply:
You're perfectly right! I changed it a bit to set as False when it is None because of the next condition

Reply:
Makes sense.

Reply:
(Or move that condition into the first branch.)